server/svix-server/src/worker.rs

```diff
@@ -291,6 +257,89 @@ fn bytes_to_string(bytes: bytes::Bytes) -> String {
     }
 }
 
+/// Manages preparation and execution of a QueueTask type
+async fn process_task(
+    cfg: Configuration,
+    db: &DatabaseConnection,
+    cache: Cache,
+    queue_tx: &TaskQueueProducer,
+    queue_task: QueueTask,
+) -> Result<()> {
+    let msg = message::Entity::find_by_id(queue_task.clone().msg_id())
+        .one(db)
+        .await?
+        .ok_or_else(|| {
+            Error::Generic(format!(
+                "Unexpected: message doesn't exist {}",
+                queue_task.clone().msg_id()
+            ))
+        })?;
+    let payload = msg.payload.as_ref().expect("Message payload is NULL");
+
+    let create_message_app = CreateMessageApp::layered_fetch(
+        cache.clone(),
+        db,
+        None,
+        msg.app_id.clone(),
+        msg.org_id.clone(),
+        Duration::from_secs(30),
+    )
+    .await?
+    .ok_or_else(|| Error::Generic(format!("Application doesn't exist: {}", &msg.app_id)))?;
+
+    let endpoints: Vec<CreateMessageEndpoint> = create_message_app
+        .filtered_endpoints(queue_task.clone().trigger_type(), &msg)
+        .iter()
+        .filter(|endpoint| match &queue_task {
+            QueueTask::MessageV1(task) => task.endpoint_id == endpoint.id,
+            QueueTask::MessageBatch(_) => true,
+        })
+        .cloned()
+        .collect();
+
+    // TODO: remove this section once destinations are obsolete
+    if matches!(queue_task, QueueTask::MessageBatch(_)) {
+        let destinations = endpoints
+            .iter()
+            .map(|endpoint| messagedestination::ActiveModel {
+                msg_id: Set(msg.id.clone()),
+                endp_id: Set(endpoint.id.clone()),
+                next_attempt: Set(Some(Utc::now().into())),
+                status: Set(MessageStatus::Sending),
+                ..Default::default()
+            });
+        messagedestination::Entity::insert_many(destinations)
+            .exec(db)
+            .await?;
+    }
+
+    let futures: Vec<_> = endpoints
+        .iter()
+        .map(|endpoint| {
+            dispatch(
+                cfg.clone(),
+                db,
+                queue_tx,
+                payload,
+                queue_task.clone().to_msg_task(endpoint.id.clone()),
```

Comment:
Why do we need the clone on queue_task?

Reply:
Unfortunately I wasn't able to figure out an alternative for this one - it's because of how it is used in this loop.

Reply:
Ok.